diff --git a/app/ui.js b/app/ui.js
index 90dc718..78171ee 100644
--- a/app/ui.js
+++ b/app/ui.js
@@ -17,6 +17,7 @@ import keysyms from "../core/input/keysymdef.js";
 import Keyboard from "../core/input/keyboard.js";
 import RFB from "../core/rfb.js";
 import * as WebUtil from "./webutil.js";
+import WsAudio from "../audio.js";
 
 const PAGE_TITLE = "noVNC";
 
@@ -1031,6 +1032,7 @@ const UI = {
         }
 
         try {
+	    UI.audio = WsAudio();
             UI.rfb = new RFB(document.getElementById('noVNC_container'),
                              url.href,
                              { shared: UI.getSetting('shared'),
@@ -1064,6 +1066,7 @@ const UI = {
     },
 
     disconnect() {
+	UI.audio.close();
         UI.rfb.disconnect();
 
         UI.connected = false;
